index.js

```diff
@@ -335,16 +335,25 @@ module.exports = function (log, indexesPath) {
       write: function (record) {
         if (updateOffsetIndex(offset, record.seq)) updatedOffsetIndex = true
 
-        if (updateTimestampIndex(offset, record.seq, record.value))
-          updatedTimestampIndex = true
-
-        if (updateSequenceIndex(offset, record.seq, record.value))
-          updatedSequenceIndex = true
-
-        if (indexNeedsUpdate) {
-          if (op.data.prefix)
-            updatePrefixIndex(op.data, index, record.value, offset, record.seq)
-          else updateIndexValue(op, index, record.value, offset)
+        if (record.value != null) {
```

Comment:
How about `if (!record.value) return // deleted record`? It reduces the amount of nested ifs, which can get hard to read

Reply:
we still need the offset++, but I guess I could do that in the return

Reply:
Yeah, I thought that maybe the initial value of offset could be `-1`, and then first thing inside the stream iteration we do is `offset++`. That's one way, I'm not saying it has to be that way